tests/suites/host_test.function

```diff
@@ -715,30 +714,33 @@ int execute_tests(int argc, const char **argv)
                 unmet_dep_count = 0;
                 missing_unmet_dependencies = 0;
             } else if (ret == DISPATCH_TEST_SUCCESS) {
-                if (mbedtls_test_info.result == MBEDTLS_TEST_RESULT_SUCCESS) {
+                if (mbedtls_test_get_result() == MBEDTLS_TEST_RESULT_SUCCESS) {
                     mbedtls_fprintf(stdout, "PASS\n");
-                } else if (mbedtls_test_info.result == MBEDTLS_TEST_RESULT_SKIPPED) {
+                } else if (mbedtls_test_get_result() == MBEDTLS_TEST_RESULT_SKIPPED) {
                     mbedtls_fprintf(stdout, "----\n");
                     total_skipped++;
                 } else {
+                    char line_buffer[MBEDTLS_TEST_LINE_LENGTH];
+
                     total_errors++;
                     mbedtls_fprintf(stdout, "FAILED\n");
                     mbedtls_fprintf(stdout, "  %s\n  at ",
-                                    mbedtls_test_info.test);
-                    if (mbedtls_test_info.step != (unsigned long) (-1)) {
+                                    mbedtls_test_get_test());
+                    if (mbedtls_test_get_step() != (unsigned long) (-1)) {
```

Comment:
Running multiple test functions at once will cause this step variable to be wrongly reported. If one thread fails at step `n`, other threads will keep writing to the step variable and in the final test report the step will likely be wrong.

To keep this step functionality we probably want a way for the first failing thread (the one whose failure is reported) to set `step` in a more permanent way. The issue with this is that calls to `mbedtls_test_set_step` are numerous and the cases are all unique, each "step" corresponds to blocks of code containing multiple checks, and other threads can change `step` at any time; I can't see an easy fix.